### DISCUSSION TITLE
Unexpected failure: Mismatch between ADT and runtime classes

### DISCUSSION THREAD

**Discussion by smartcodegames** [airsdk/Adobe-Runtime-Support#731] [Q&A] [answered]
This error comes up building for Android with the latest AIR SDK (33.1.1.406) and JRE 1.8.0_112. I use this java version because I have an old 1024bit certificate, and it's the latest version that supports it.
I'm using a custom adt.jar sent by @ajwfrost because of a memory problem in #696

**Comment by ajwfrost** [comment, marked as answer]
Hi - this mismatch happens if you've got an SDK that's a little mixed up.. there's a version code built into ADT and there's a version of the Android libraries that are in the SDK, and if they don't match then it can end up with incompatibilities and linker errors later, hence this initial check...

The test version of ADT that I sent I thought was built with the 33.1.1.406 version code though, so if you're using that on top of the normal 33.1.1.406 SDK it should be okay?

**Reply by smartcodegames** [reply]
You're right, we had different versions, here is adt.log:
Version in APK = 33.1.1.345, version of ADT = 33.1.1.406
Unexpected failure: Mismatch between ADT and runtime classes

Installed AIR 33.1.1.406, and it compiles! Animate is misleading, because it displays the version number based upon the current adt.jar, not the installed version so I saw v406 in the editor. Thanks a lot!
